art/estimators/certification/derandomized_smoothing/derandomized_smoothing.py

```diff
@@ -0,0 +1,445 @@
+# SOFTWARE.
+"""
+This module implements (De)Randomized Smoothing for Certifiable Defense against Patch Attacks
+
+| Paper link: https://arxiv.org/abs/2002.10733
+"""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from abc import ABC, abstractmethod
+
+from typing import Optional, Union
+import random
+
+import numpy as np
+
+
+class DeRandomizedSmoothingMixin(ABC):
+    """
+    Implementation of (De)Randomized Smoothing applied to classifier predictions as introduced
+    in Levine et al. (2020).
+
+    | Paper link: https://arxiv.org/abs/2002.10733
+    """
+
+    def __init__(
+        self,
+        ablation_type: str,
+        ablation_size: int,
+        threshold: float,
+        logits: bool,
+        channels_first: bool,
+        *args,
+        **kwargs,
+    ) -> None:
+        """
+        Create a derandomized smoothing wrapper.
+
+        :param ablation_type: The type of ablations to perform. Currently must be either "column", "row", or "block"
+        :param ablation_size: Size of the retained image patch.
+                              An int specifying the width of the column for column ablation
+                              Or an int specifying the height/width of a square for block ablation
+        :param threshold: The minimum threshold to count a prediction.
+        :param logits: if the model returns logits or normalized probabilities
+        :param channels_first: If the channels are first or last.
+        """
+        super().__init__(*args, **kwargs)  # type: ignore
+        self.ablation_type = ablation_type
+        self.logits = logits
+        self.threshold = threshold
+        self._channels_first = channels_first
+        self.ablator: ABLATOR_TYPE
+
+        if self.ablation_type in {"column", "row"}:
+            row_ablation_mode = self.ablation_type == "row"
+            self.ablator = ColumnAblator(
+                ablation_size=ablation_size, channels_first=self._channels_first, row_ablation_mode=row_ablation_mode
+            )
+        elif self.ablation_type == "block":
+            self.ablator = BlockAblator(ablation_size=ablation_size, channels_first=self._channels_first)
+        else:
+            raise ValueError("Ablation type not supported. Must be either column or block")
+
+    def _predict_classifier(self, x: np.ndarray, batch_size: int, training_mode: bool, **kwargs) -> np.ndarray:
+        """
+        Perform prediction for a batch of inputs.
+
+        :param x: Input samples.
+        :param batch_size: Size of batches.
+        :param training_mode: `True` for model set to training mode and `'False` for model set to evaluation mode.
+        :return: Array of predictions of shape `(nb_inputs, nb_classes)`.
+        """
+        raise NotImplementedError
+
+    def predict(self, x: np.ndarray, batch_size: int = 128, **kwargs) -> np.ndarray:  # pylint: disable=W0613
```

Comment:
Which argument is unused causing W0613?

Reply:
Hi Beat, the **kwargs argument is unused to keep the same function signature as other predict methods.

Reply:
Ok, I see. How about if we pass `kwargs` to the calls of _predict_classifier`?